lib/utils.js

```diff
@@ -444,13 +520,57 @@ const POOL = new BufferPool(4096);
  */
 class Tap {
   constructor (buf, pos) {
-    this.buf = buf;
+    this.setData(buf, pos);
+  }
+
+  setData (buf, pos) {
+    if (typeof Buffer === 'function' && buf instanceof Buffer) {
+      buf = new Uint8Array(buf.buffer, buf.byteOffset, buf.length);
+    }
+    this.arr = buf;
     this.pos = pos | 0;
+    this.length = buf.length;
     if (this.pos < 0) {
       throw new Error('negative offset');
     }
   }
 
+  reinitialize (capacity) {
+    this.setData(new Uint8Array(capacity));
+  }
+
+  static fromBuffer (buf, pos) {
+    return new Tap(buf, pos);
+  }
+
+  static withCapacity (capacity) {
+    let buf = new Uint8Array(capacity);
+    return new Tap(buf);
+  }
+
+  toBuffer () {
```

Comment:
Definitely something that should be done before a breaking 6.0 release. I contemplated doing it as part of this PR, but I'm not sure what terminology would be best. `toBinary` sounds like it could refer to a binary string. Maybe `toTypedArray` or `toBytes`? Maybe rename `Type#encode` to `Type#encodeInto` a la `TextEncoder` and repurpose the old method name?

Reply:
An "Into" suffix for `encode` sounds good; I like the consistency with `TextEncoder`. `binaryEncode` (and `jsonEncode`) might be fine, they mirror the wording in the [Avro specification](https://avro.apache.org/docs/1.11.1/specification/#data-serialization-and-deserialization).